diff --git a/src/Version.h b/src/Version.h
index e5af9dc..6004f73 100644
--- a/src/Version.h
+++ b/src/Version.h
@@ -5,7 +5,7 @@
 const char* getVersion();
 const char* getAgent();
 
-enum class Platform { Windows_x64, Windows_arm64, Linux_x64, Linux_arm64, MacOS_arm64 };
+enum class Platform { Windows_x64, Windows_arm64, Linux_x64, Linux_arm64, MacOS_x64, MacOS_arm64 };
 
 #if defined(_WIN64) && defined(_M_AMD64)
 constexpr Platform PRD_CURRENT_PLATFORM = Platform::Windows_x64;
@@ -16,6 +16,8 @@ constexpr Platform PRD_CURRENT_PLATFORM = Platform::Linux_x64;
 constexpr Platform PRD_CURRENT_PLATFORM = Platform::Linux_arm64;
 #elif defined(__APPLE__) && defined(__aarch64__)
 constexpr Platform PRD_CURRENT_PLATFORM = Platform::MacOS_arm64;
+#elif defined(__APPLE__) && defined(__x86_64__)
+constexpr Platform PRD_CURRENT_PLATFORM = Platform::MacOS_x64;
 #endif
 
 const char* platformToString(Platform Platform);
diff --git a/src/Version.cpp b/src/Version.cpp
index 378af94..ad029a9 100644
--- a/src/Version.cpp
+++ b/src/Version.cpp
@@ -34,6 +34,8 @@ const char* platformToString(Platform platform)
 			return "windows64";
 		case Platform::Windows_arm64:
 			return "windows_arm64";
+		case Platform::MacOS_x64:
+			return "macos64";
 		case Platform::MacOS_arm64:
 			return "macos_arm64";
 		case Platform::Linux_arm64:
diff --git a/src/pr-downloader.cpp b/src/pr-downloader.cpp
index b4551a0..1fd2208 100644
--- a/src/pr-downloader.cpp
+++ b/src/pr-downloader.cpp
@@ -49,6 +49,8 @@ DownloadEnum::Category getPlatformEngineCat()
 			return DownloadEnum::CAT_ENGINE_WINDOWS_ARM64;
 		case Platform::Linux_arm64:
 			return DownloadEnum::CAT_ENGINE_LINUX_ARM64;
+		case Platform::MacOS_x64:
+			return DownloadEnum::CAT_ENGINE_MACOSX;
 		case Platform::MacOS_arm64:
 			return DownloadEnum::CAT_ENGINE_MACOSX_ARM64;
 		default:
